src/resonate.ts

```diff
@@ -134,25 +134,24 @@ export class Resonate {
   /**
    * Register a function and returns a registered function
    */
-  public register<F extends Func>(name: string, func: F): ResonateFunc<F>;
-  public register<F extends Func>(func: F): ResonateFunc<F>;
-  public register<F extends Func>(nameOrFunc: string | F, maybeFunc?: F): ResonateFunc<F> {
-    const name = typeof nameOrFunc === "string" ? nameOrFunc : nameOrFunc.name;
-    const func = typeof nameOrFunc === "string" ? maybeFunc! : nameOrFunc;
-
-    if (this.registry.has(name)) {
-      throw new Error(`'${name}' already registered`);
-    }
+  public register<F extends Func>(name: string, func: F, version?: number): ResonateFunc<F>;
```

Comment:
Can we change the signature of this function to
```
public register<F extends Func>(name: string, func: F, { version?: number } = { version: 1 })
```

I anticipate the ability to set certain options on register (we used to support this) so throwing the final argument into a dict will help us maintain backwards compatibility once we get there.